src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,653 @@
+    function accrueVcon(address venue) external globalLock(1) whenNotPaused {
+        IPCVOracle oracle = pcvOracle();
+
+        require(oracle.isVenue(venue), "MarketGovernance: invalid destination");
+
+        _accrue(venue);
+    }
+
+    /// ---------- Permissionless User PCV Allocation Methods ----------
+
+    /// @notice a user can get slashed up to their full VCON stake for entering
+    /// a venue that takes a loss.
+    /// @param amountVcon to stake on destination
+    /// @param destination address to accrue rewards to, and send funds to
+    function stake(
+        uint256 amountVcon,
+        address destination
+    ) external globalLock(1) whenNotPaused {
+        IPCVOracle oracle = pcvOracle();
+        require(
+            oracle.isVenue(destination),
+            "MarketGovernance: invalid destination"
+        );
+
+        _accrue(destination); /// update share price in the destination so the user gets in at the current share price
+
+        /// vconToShares will always return correctly as accrue will set venueLastRecordedVconSharePrice
+        /// to the correct share price from 0 if uninitialized
+        uint256 userShareAmount = vconToShares(destination, amountVcon);
+
+        /// user updates
+        venueUserShares[destination][msg.sender] += userShareAmount;
+
+        /// venue updates
+        venueTotalShares[destination] += userShareAmount;
+
+        /// check and an interaction with a trusted contract
+        vcon().safeTransferFrom(msg.sender, address(this), amountVcon); /// transfer VCON in
+
+        emit Staked(destination, msg.sender, amountVcon);
+    }
+
+    /// @notice unstake VCON and transfer corresponding VCON to another venue
+    /// @param shareAmount the amount of shares to unstake
+    /// @param source address to accrue rewards to, and pull funds from
+    /// @param vconRecipient address to receive the VCON
+    /// @dev both source and destination are checked twice,
+    /// the first time in the market governance contract,
+    /// the second time in the PCV Router contract.
+    function unstake(
+        uint256 shareAmount,
+        address source,
+        address vconRecipient
+    ) external globalLock(1) whenNotPaused {
+        /// ---------- Checks ----------
+        IPCVOracle oracle = pcvOracle();
+
+        require(oracle.isVenue(source), "MarketGovernance: invalid source");
+        require(
+            venueUserShares[source][msg.sender] >= shareAmount,
+            "MarketGovernance: invalid share amount"
+        );
+
+        address denomination = IPCVDepositV2(source).balanceReportedIn();
+        address destination = underlyingTokenToHoldingDeposit[denomination];
+        require(
+            destination != address(0),
+            "MarketGovernance: invalid destination"
+        );
+
+        /// ---------- Effects ----------
+
+        _accrue(source); /// update profitPerVCON in the source so the user gets paid out at the current share price
```

Comment:
This will revert if the venue suffered from a loss, so users can't unstake from a venue unless governance applied the loss, am I getting this right ? that doesn't seem optimal, we want vcon stakers in the lossy venue to move as much pcv out of the venue as possible

Reply:
We don't anticipate that any of these lossless yield venues will ever experience a loss through a markdown in share prices as they are all (compound, aave, euler, morpho, DSR) designed to increase or stay constant in their share price. This means that VCON stakers can pull funds from a venue while there is a loss, provided they can remove their pro-rata share of PCV, because that loss will very likely not be detected by the underlying venue. In the fuse and cream hacks, CToken share prices didn't mark down even though the CTokens become worthless.